File: aat-android/src/main/java/ch/bailu/aat/gpx/attributes/SensorInformation.java
package ch.bailu.aat.gpx.attributes;

import ch.bailu.aat_lib.gpx.GpxInformation;
import ch.bailu.aat_lib.gpx.attributes.GpxAttributes;

public final class SensorInformation extends GpxInformation {
        private final GpxAttributes attributes;
        private final long timeStamp = System.currentTimeMillis();


        public SensorInformation(GpxAttributes a) {
            attributes = a;
        }

        @Override
        public GpxAttributes getAttributes() {
            return attributes;
        }

        @Override
        public long getTimeStamp() {
            return timeStamp;
        }
    }

File: aat-lib/src/main/java/ch/bailu/aat_lib/gpx/attributes/SensorInformation.kt
package ch.bailu.aat_lib.gpx.attributes

import ch.bailu.aat_lib.gpx.information.GpxInformation

/** Wraps a [GpxAttributes] as a [GpxInformation] snapshot, timestamped at construction time. */
class SensorInformation(private val attributes: GpxAttributes) : GpxInformation() {
    private val timeStamp = System.currentTimeMillis()
    override fun getAttributes(): GpxAttributes {
        return attributes
    }

    override fun getTimeStamp(): Long {
        return timeStamp
    }
}
